### PR TITLE
ISSUE-86 Expose a EmailGetSimulation for JMAP-RFC-8621

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapEmail.scala [hunk 1]
```diff
@@ -25,4 +25,27 @@ object JmapEmail {
 
   val nonEmptyListMessagesChecks: HttpCheck =
     jsonPath("$.methodResponses[0][1].ids[*]").findAll.saveAs("emailIds")
+
+  def getRandomEmails(queryParameters: JmapParameters = NO_PARAMETERS): HttpRequestBuilder = {
```
Comment (chibenwa) [inline src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapEmail.scala]: I bet you do not get a 'single' random email, but rather get them all...

To select a random email please stick to what is done on draft ():

 - 1. Do a EmailQuery first and save the returned Ids in the session (emailIds variable) - please do that as a separate spec as well - it enable a finer grain diagnostic.
 - 2. Then use the same syntax as `JmapMessages::getRandomMessages`: `"ids": ["$${$emailIds.random()}"],`

Comment (Arsnael) [reply src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapEmail.scala]: ok with your comment I finally understood how it works... The query part is done in `CommonSteps.provisionUsersWithMessageList`on draft, I missed that one thanks...

Comment (chibenwa) [reply src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapEmail.scala]: I would be in favor of moving the query part out of the provisionning. 

To make it explicit...

File: src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/scenari/EmailGetScenario.scala [hunk 2]
```diff
@@ -14,11 +14,14 @@ class EmailGetScenario {
...
     scenario("EmailGetScenario")
       .feed(userFeeder)
-      .exec(CommonSteps.provisionUsersWithMessageList(recipientFeeder, numberOfMessages = 10))
+      .exec(CommonSteps.provisionUsersWithMessages(recipientFeeder, numberOfMessages = 10))
       .exec(SessionStep.retrieveAccountId)
       .during(duration) {
-        exec(JmapEmail.getRandomEmails()
-          .check(JmapHttp.statusOk, JmapHttp.noError, JmapEmail.nonEmptyListMessagesChecks))
+        exec(JmapEmail.queryEmails()
+            .check(JmapHttp.statusOk, JmapHttp.noError, JmapEmail.nonEmptyListMessagesChecks))
```
Comment (chibenwa) [inline src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/scenari/EmailGetScenario.scala:21]: You don't save the email (id) list in the session here?

Comment (Arsnael) [reply src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/scenari/EmailGetScenario.scala:21]: It's done in the `JmapEmail.nonEmptyListMessagesChecks`, that you copied from draft?

**Review by chibenwa — APPROVED** [review]

File: src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/scenari/EmailGetScenario.scala [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import io.gatling.http.Predef._
+import org.apache.james.gatling.control.RecipientFeeder.RecipientFeederBuilder
+import org.apache.james.gatling.control.UserFeeder.UserFeederBuilder
+import org.apache.james.gatling.jmap.draft.CommonSteps
+import org.apache.james.gatling.jmap.rfc8621.{JmapEmail, JmapHttp, SessionStep}
+
+import scala.concurrent.duration._
+
+class EmailGetScenario {
+  def generate(duration: Duration, userFeeder : UserFeederBuilder, recipientFeeder: RecipientFeederBuilder): ScenarioBuilder =
```
Comment (LanKhuat) [inline src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/scenari/EmailGetScenario.scala]: extra space

**Review by LanKhuat — APPROVED** [review]

**Pull request merged**